### ISSUE TITLE
Any interest in a prettier README?

### ISSUE THREAD

**Issue by danielquinn** [jstpcs/lnxpcs#60]
Your primary README is growing with each new graphic and I thought it might be nice to have a git hook or something that dynamically generates a README (or series of READMEs) with images and links to shirts etc.

I've got half of the code to do it (Python3) but won't continue if there's no interest.  What do you think?

Something like this, but with links to shirts & whatever else:

![screenshot from 2018-02-08 12-58-32](https://user-images.githubusercontent.com/218205/35974163-d4f9f048-0ccf-11e8-9e8e-d7380b54691d.png)

**Comment by danielquinn** [comment]
Ooh!  Here's the dynamically-generated markup (better than a screenshot):

|       |       |       |
| :---: | :---: | :---: |
| ![arch-card](https://raw.githubusercontent.com/jstpcs/lnxpcs/master/cards/classic/arch-card.png) | ![tor-card](https://raw.githubusercontent.com/jstpcs/lnxpcs/master/cards/classic/tor-card.png) | ![gnu-card](https://raw.githubusercontent.com/jstpcs/lnxpcs/master/cards/classic/gnu-card.png) |
| ![manjaro-card](https://raw.githubusercontent.com/jstpcs/lnxpcs/master/cards/classic/manjaro-card.png) | ![solus-card](https://raw.githubusercontent.com/jstpcs/lnxpcs/master/cards/classic/solus-card.png) | ![python-card](https://raw.githubusercontent.com/jstpcs/lnxpcs/master/cards/classic/python-card.png) |
| ![linuxmasterrace-card](https://raw.githubusercontent.com/jstpcs/lnxpcs/master/cards/classic/linuxmasterrace-card.png) | ![php-card](https://raw.githubusercontent.com/jstpcs/lnxpcs/master/cards/classic/php-card.png) | ![gnu-linux-card](https://raw.githubusercontent.com/jstpcs/lnxpcs/master/cards/classic/gnu-linux-card.png) |
| ![sudo-card](https://raw.githubusercontent.com/jstpcs/lnxpcs/master/cards/classic/sudo-card.png) | ![iptables-card](https://raw.githubusercontent.com/jstpcs/lnxpcs/master/cards/classic/iptables-card.png) | ![arch-card-blue](https://raw.githubusercontent.com/jstpcs/lnxpcs/master/cards/classic/arch-card-blue.png) |
| ![userdel-card](https://raw.githubusercontent.com/jstpcs/lnxpcs/master/cards/classic/userdel-card.png) | ![cron-card](https://raw.githubusercontent.com/jstpcs/lnxpcs/master/cards/classic/cron-card.png) | ![kill-card](https://raw.githubusercontent.com/jstpcs/lnxpcs/master/cards/classic/kill-card.png) |
| ![swift-card](https://raw.githubusercontent.com/jstpcs/lnxpcs/master/cards/classic/swift-card.png) | ![fedora-card](https://raw.githubusercontent.com/jstpcs/lnxpcs/master/cards/classic/fedora-card.png) | ![debian-card](https://raw.githubusercontent.com/jstpcs/lnxpcs/master/cards/classic/debian-card.png) |
| ![freebsd-card](https://raw.githubusercontent.com/jstpcs/lnxpcs/master/cards/classic/freebsd-card.png) | ![kali-card](https://raw.githubusercontent.com/jstpcs/lnxpcs/master/cards/classic/kali-card.png) | ![root-card](https://raw.githubusercontent.com/jstpcs/lnxpcs/master/cards/classic/root-card.png) |

**Comment by danielquinn** [comment]
That's what I was thinking, something like a separate README for each directory containing previews of all the images.  It'd be nice to include links to the shirts and other stuff too to help @jstpcs make some money of of their excellent work ;-)

**Comment by jstpcs** [comment]
hi! that looks really amazing. i made a simple [site](linux.pictures) with all the previews. but having the such kind of previews directly in the repo would be great.

**Comment by danielquinn** [comment]
Yeah I like the site, but the problem is that this Github repo is likely to be a lot of people's first time seeing this stuff, so it'd be nice to have something a little more visual.

@jstpcs: a question about the images on your site: are they following a strict naming convention from the source files?  I ask because in the example here, the files aren't thumbnails, but the original, which can be *brutal* on a slow connection.  It would be nice to be able to reference lower-res thumbnails for this but auto-generating them seemed a bit nutty for a commit hook.  If the thumbnails on your site follow the same naming convention as the source images, I can patch the README generator to use those images instead... assuming Github allows for offsite image references.

**Comment by jstpcs** [comment]
@danielquinn, thanks a lot for your efforts to help. frankly speaking, if you can make the previews here, may be i can get rid of the site at all, so i'd prefer not to reference images outside the github. i can generate the thumbs for all the images and put them in separate folders. so in every folder that contains images there would be a separate folder **thumbs** with previews. these images can follow a strict name convention - for example for image arch-lines.png the thumb will be ./thumbs/arch-lines-thumb.png.

**Comment by danielquinn** [comment]
Sure no problem.  I have some code based on Pillow that will do the job.  The only problem is that I don't know how to automate the process as a git hook, so you may have to run it manually and commit the results.

**Comment by jstpcs** [comment]
@danielquinn, that's not a problem at all to run the script after i create a new pic to regenerate the readme.

**Comment by danielquinn** [comment]
Alright then!  I'll try to spend some time this week adding thumbnailing to the script and will post back when I have something.
